metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -308,6 +328,48 @@ private ApiError validateAlterConfig(ConfigResource configResource,
         return ApiError.NONE;
     }
 
+    void maybeTriggerPartitionUpdateOnMinIsrChange(List<ApiMessageAndVersion> records) {
+        List<ConfigRecord> minIsrRecords = new ArrayList<>();
+        Map<String, String> topicMap = new HashMap<>();
+        Map<String, String> configRemovedTopicMap = new HashMap<>();
+        records.forEach(record -> {
+            if (MetadataRecordType.fromId(record.message().apiKey()) == MetadataRecordType.CONFIG_RECORD) {
+                ConfigRecord configRecord = (ConfigRecord) record.message();
+                if (configRecord.name().equals(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG)) {
+                    minIsrRecords.add(configRecord);
+                    if (Type.forId(configRecord.resourceType()) == Type.TOPIC) {
+                        if (configRecord.value() == null) topicMap.put(configRecord.resourceName(), configRecord.value());
+                        else configRemovedTopicMap.put(configRecord.resourceName(), configRecord.value());
+                    }
+                }
+            }
+        });
```

Comment:
what is the behavior if the default broker config for `min.insync.replicas` is changed? 
I am not actually sure how that impacts the `min.insync.replicas` for existing topics.

Reply:
If `min.insync.replicas` is not set on the topic config level, the effective `min.insync.replicas` of a topic will change if default broker config is updated.

Reply:
Yeah, unfortunately we have a 5 level system:

1. topic configs (highest priority)
2. node configs (aka "broker" configs, but they also apply to controllers)
3. cluster configs (aka default configs for the broker resource)
4. static configuration
5. static default

The first three levels can change at runtime 🤮